libs/ml/include/ml/ops/ops.hpp

```diff
@@ -73,8 +95,128 @@ class Ops
     return is_training_;
   }
 
+  virtual void SetSliceOutputShape(Shape const &new_shape)
```

Comment:
I don't like all of these XSliceX names. I understand why you're saying slice, but in general a slice can be across any dimension; you're talking about a very particular way of slicing - batch slicing. Maybe these should all be called SetBatchOutputShape, ComputeBatchOutputShape etc.?

Reply:
Batch is a nice name too!